src/Rings/integer-localizations.jl

```diff
@@ -0,0 +1,197 @@
+
+### required functionality
+function Base.in(a::fmpz, S::FmpzPowersOfElement) 
+  # check whether ∃ c ∈ ℤ, k ∈ ℕ₀: c ⋅ a = (∏ᵢ dᵢ)ᵏ, where dᵢ are the admissible denominators in S.
+  d = prod(denominators(S))
+  g = gcd(a, d)
+  while(!(g==one(ZZ)))
+    a = divexact(a, g)
+    g = gcd(a, d)
+  end
+  return a == one(ZZ)
+end
+
+### additional constructors
+FmpzPowersOfElement(d::fmpz) = FmpzPowersOfElement([d])
+FmpzPowersOfElement(d::Int) = FmpzPowersOfElement(ZZ(d))
+
+### additional functionality
+denominators(S::FmpzPowersOfElement) = S.d::Vector{fmpz}
+
+
+#################################################################################
+# Complements of prime ideals in ℤ                                              #
+#################################################################################
+
+@Markdown.doc """
+    FmpzComplementOfPrimeIdeal
+
+The multiplicatively closed set `S = ℤ ∖ ⟨p⟩` of integers outside a prime ideal `⟨p⟩`.
+"""
+mutable struct FmpzComplementOfPrimeIdeal <: AbsComplementOfPrimeIdeal{FlintIntegerRing, Hecke.ZZIdl, fmpz}
+  # essential fields
+  p::fmpz
+
+  function FmpzComplementOfPrimeIdeal(p::fmpz)
+    isprime(p) || error("$(p) is not prime")
+    return new(p)
+  end
+end
+
+### required getter functions
+ambient_ring(S::FmpzComplementOfPrimeIdeal) = ZZ
+
+### required functionality
+function Base.in(b::fmpz, S::FmpzComplementOfPrimeIdeal)
+  return !(divides(b, S.p)[1])
+end
+
+### additional constructors
+FmpzComplementOfPrimeIdeal(i::Int) = FmpzComplementOfPrimeIdeal(ZZ(i))
+
+
+#################################################################################
+# The complement of the zero ideal in ℤ                                         #
+#################################################################################
+
+@Markdown.doc """
+    FmpzComplementOfZeroIdeal <: AbsComplementOfPrimeIdeal{FlintIntegerRing, Hecke.ZZIdl, fmpz}
+
+The complement of the zero ideal in `ℤ`.
+"""
+mutable struct FmpzComplementOfZeroIdeal <: AbsComplementOfPrimeIdeal{FlintIntegerRing, Hecke.ZZIdl, fmpz}
+  function FmpzComplementOfZeroIdeal() 
+    return new{}()
+  end
+end
+
+### required getter functions
+ambient_ring(S::FmpzComplementOfZeroIdeal) = ZZ
+
+### required functionality
+function Base.in(b::fmpz, S::FmpzComplementOfZeroIdeal) 
+  return !(b == zero(ZZ))
+end
+
+
+#################################################################################
+# Localizations of ℤ                                                            #
+#################################################################################
+
+@Markdown.doc """
+    FmpzLocalizedRing{MultSetType} <: AbsLocalizedRing{FlintIntegerRing, fmpz, MultSetType} 
+
+A minimal implementation for the localization `ℤ[S⁻¹]` of the ring of integers 
+at a multiplicatively closed set `S` of type `MultSetType`.
+"""
+mutable struct FmpzLocalizedRing{MultSetType} <: AbsLocalizedRing{FlintIntegerRing, fmpz, MultSetType} 
```

Comment:
You could also have the type restriction you manually impose in the constructor directly in the type signature:
```suggestion
mutable struct FmpzLocalizedRing{MultSetType <: AbsMultSet{FlintIntegerRing, fmpz}} <: AbsLocalizedRing{FlintIntegerRing, fmpz, MultSetType} 
```